app/controllers/user/UserProfile.controller.ts

```diff
@@ -5,6 +5,62 @@ import UserProfile from '../../models/UserProfile';
 import { IProfileRequest } from '../../request/IProfileRequest';
 import { IUserRequest } from '../../request/IRequest';
 
+/**
+ * @api {get} /users/:userId/profile Get all of the profile information for a certain user.
+ * @apiName ShowProfileforUser
+ * @apiGroup User
+ * 
+ * @apiParam {string} username A partial or full username for a given user.
```

Comment:
id, the id is used as the parameter within the given url. Never usernames since these can change.